hyper/http11/connection.py

```diff
@@ -192,6 +204,23 @@ def request(self, method, url, body=None, headers=None):
 
         return
 
+    def _headers_to_http_header_map(self, headers):
+        headers = headers or {}
+        if not isinstance(headers, HTTPHeaderMap):
+            if isinstance(headers, Mapping):
+                headers = HTTPHeaderMap(headers.items())
+            elif isinstance(headers, Iterable):
+                headers = HTTPHeaderMap(headers)
+            else:
+                raise ValueError(
+                    'Header argument must be a dictionary or an iterable'
+                )
+        return headers
```

Comment:
So should it be moved to the top of the file as a function rather than a method of the class? Or maybe even as a classmethod (?) of `HTTPHeaderMap` itself?

BTW, http11 connection accepts headers as Iterable, but http2 connection assumes that headers is always a Mapping (having `.items()` method).

I guess, for clarity, either http20 should use the same function as this one or http11 should also assume that headers is a Mapping.